File: backend/src/main/java/com/library/backend/api/bookloan/interfaces/BookLoanService.java
package com.library.backend.api.bookloan.interfaces;

import com.library.backend.api.bookloan.data.BookLoanRequest;
import com.library.backend.api.bookloan.data.BookLoanResponse;
import com.library.backend.entities.user.User;

import java.util.List;

public interface BookLoanService {

    BookLoanResponse addNewLoan(long bookId, User user);

    List<BookLoanResponse> getLoansForGivenUser(String username, User user);

    BookLoanResponse returnBook(long loanId, User user);
}

File: frontend/src/app/core/services/bookloans/book-loan.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {BookLoan} from "../../data/bookloans/book-loan";
import {ApiUrl} from "../../enums/api-url";
import {environment} from "../../../../environments/environment";

@Injectable({
  providedIn: 'root'
})
export class BookLoanService {

  constructor(private http: HttpClient) { }

  borrowBook(bookId: number) {
    return this.http.post<BookLoan>(`${environment.API_URL}${ApiUrl.BOOK_LOANS_ADD}`, null,{params: {bookId: bookId}});
  }

  getBookLoans(username: string) {
    return this.http.get<BookLoan[]>(`${environment.API_URL}${ApiUrl.BOOK_LOANS_GET_ALL}`, {params: {username: username}});
  }

  returnBook(bookLoanId: number) {
    return this.http.post<BookLoan>(`${environment.API_URL}${ApiUrl.BOOK_LOANS_RETURN}`, null ,{params: {loanId: bookLoanId}});
  }
}
